web/src/app/shared/configuration/field.service.ts

```diff
@@ -183,34 +217,21 @@ export class FieldService {
   /**
    * Filter by group and all fields
    */
-  public filterApply(dataOptions: itemOptions[], c: { advanced: boolean; search: string }): itemOptions[] {
-    return dataOptions.filter((a) => this.isVisibleField(a)).map((a) => this.handleTree(a, c));
-  }
-
-  private handleTree(a: itemOptions, c: { advanced: boolean; search: string }) {
-    if ('options' in a) {
-      const result = a.options.map((b) => this.handleTree(b, c));
-      if (c.search) a.hidden = a.options.filter((b) => !b.hidden).length === 0;
-      else a.hidden = this.isAdvancedField(a) ? !c.advanced : false;
-      return result;
-    } else if (this.isVisibleField(a)) {
-      a.hidden = !(a.display_name.toLowerCase().includes(c.search.toLowerCase()) || JSON.stringify(a.value).includes(c.search));
-      if (!a.hidden && this.isAdvancedField(a)) a.hidden = !c.advanced;
-      return a;
-    }
+  public filterApply(options: TFormOptions[], c: ISearchParam): TFormOptions[] {
+    return options.filter((a) => isVisibleField(a)).map(handleTree(c));
   }
 
   /**
    * Output form, cast to source type
    */
   public parseValue(output: IOutput, source: ISource[]): IOutput {
-    const findField = (name: string, p?: string): Partial<FieldStack> => source.find((a) => (p ? a.name === p && a.subname === name : a.name === name) && !a.read_only);
+    const findField = (name: string, p?: string): Partial<IFieldStack> => source.find((a) => (p ? a.name === p && a.subname === name : a.name === name) && !a.read_only);
 
-    const runYspecParse = (v: any, f: Partial<FieldOptions>) => ((v === {} || v === []) && !f.value ? f.value : this.runYspec(v, f.limits.rules));
+    const runYspecParse = (v: any, f: Partial<IFieldOptions>) => ((v === {} || v === []) && !f.value ? f.value : this.runYspec(v, f.limits.rules));
```

Comment:
It seems to me this row have a bug

Reply:
1). any "v" !== {} and !== []
2). "(v === {} || v === [])" === false
3). "(v === {} || v === []) && !f.value" always === false

Reply:
fixed